rfc/2025-07-22-conditional-enable.md

```diff
@@ -0,0 +1,130 @@
+# Conditional field for enabling resources and modules
+
+## Current status
+
+Issue: https://github.com/opentofu/opentofu/issues/1306
+
+Right now, OpenTofu supports conditional enable/disable of resources by using a workaround with `count`.
+This approach brings a few problems, like adding indexes to resources that would be a single instance, making it harder to manage using these indexes.
+
+This RFC proposes a new way to do a cleaner and semantic way by using a new field on the `lifecycle` block called `enabled`:
+
+```
+resource "aws_instance" "example" {
+  # ...
+
+  lifecycle {
+    enabled = var.enable_server
+  }
+}
+```
+
+This is available not only to resources, but to modules too:
+
+```
+module "modcall" {
+  source = "./mod"
+
+  lifecycle {
+    enabled = var.env == "production"
+  }
+}
+```
+
+## Proposed Solution
+
+1. Raise errors if a resource is used while being disabled;
+1. Support for conditional enabling on:
+  1. Resources - Add a new field on the `lifecycle` block called `enabled`;
```

Comment:
Why not data sources?

Reply:
`resource`, `data`, and `ephemeral` blocks all declare "resources" really. It's just an annoyance of history that the first kind of resource -- managed resources -- squatted on the generic block type name before anyone realized that there would be resources of other kinds. 

So this item is talking about resources of all kinds, just as with `count` and `for_each`. The same code handles all three when it comes to these meta-arguments.

(I don't intend this comment to mean that it would not still be worth changing the RFC texr to make this clearer. I suggest we directly name all of the block types that this new argument can appear in, to make it precise: `resource`, `data`, `ephemeral` and `module`.)